docs/comparisons.md

```diff
@@ -111,3 +111,19 @@ virtualenvs with their dependencies installed.
 - fades can only run individual script files while pipx can also run packages.
 
 Example interaction: None.
+
+## pipx vs pae/pactivate
+
+_pae_ is a Bash command-line function distributed with [pactivate](https://github.com/cynic-net/pactivate) that uses pactivate to create non-ephermeral environments that can be used for both command-line applications and libraries.
+
+- pae and pipx both create isolated environments without having to specify (and remember) a directory in which to store them.
+- pae allows you to use any Python interpreter available on your system, so you can use arbitrary versions of Python.
+- pae does not require you to have anything other than a Python interpreter available; it downloads and bootstraps `pip` and `virtualenv`, always using the latest releases of both.
+- pae and pipx both make it simple to run commands in the isolated environments, but pae does not add these commands to your PATH by default. You must use `pae COMMAND` or `pae ENV COMMAND` to run them, or `pae -a ENV` to activate the environment.
+- pae allows you to activate and deactivate an isolated environment so you you can use libraries within it.
+- With pae, packages must be manually installed in an environment with `pae ENV pip install ...`.
+- pae does not create temporary environments; they must be manually created with `pae -c` and removed with `pae --rm` when no longer needed.
+- pae is significantly slower to create virtual environments due to the bootstrap process.
```

Comment:
Is there a benchmark you can link here?

Reply:
I've not done any formal benchmarking, but I did some quick comparisons. The very first time you use pipx to install something it takes about the same time as pae. But after you've done your first pipx install and `.local/share/pipx/shared` is set up, pipx time is the same as just the `pip install packagename` time, about 8 seconds faster on my slowish desktop machine.

But I'm open to your suggestions about whether this note should even be included. I personally feel it's a fairly notable difference, but then again I'm tending to create/delete virtual environments a lot with pae; other pae users that simply install more programs into existing virtual environments wouldn't have such a difference.

Reply:
If not based on measurements, can you quickly outline the technical reason why pipx is faster in this case compared to pae?

Reply:
`pipx install` is usually faster (with the notable exception of the first run) because once `.local/share/pipx/shared/` is set up it uses (as far as I can tell) the existing pip and venv there to set up a new virtual environment. `pae -c`, by contrast, always bootstraps with a newly downloaded version of pip and virtualenv before creating the actual virtual environment.

I'm not going to update this PR with this yet, because the deeper I get into examining the design decisions of pipx and comparing them with my own for pae, the more interesting things get, and the more I feel the scope of a write-up on this is exceeding what should be on the comparison page. So my tentative plan at the moment is to do a much more detailed comparison write-up for the pae documentation, and then reduce my proposed entry `docs/comparison.md` to a summary of that, with a link to the full version. (I'm open to other ideas about this, of course.)